### PR TITLE
[FLINK-4395][cep] Eager processing of late arrivals in CEP operator

### PR DESCRIPTION
**Pull request by mushketyk**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.
- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)
- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added
- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by chermenin** [comment]
@mushketyk Could you rebase this PR, please?

**Comment by mushketyk** [comment]
Hi @chermenin ,

Sure, I'll do it tomorrow.

**Comment by mushketyk** [comment]
Hi @chermenin,

To avoid rebasing this PR multiple times I'll wait until Till or another core reviewer will be available to review and merge this pull request.

Feel free to review it though.

File: flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/ProcessingType.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+package org.apache.flink.cep.operator;
+
+/**
+ * Defines what time should be considered when even is being processed.
+ */
+public enum ProcessingType {
+	// Consider time of event occurance when processing an event
+	EVENT_TIME,
+	processingType, // Consider local system time when processing an event
```
Comment (chermenin) [inline flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/ProcessingType.java]: I think this value is unnecessary.

**Comment by kl0u** [comment]
Hi @mushketyk,
Could you rebase this on the master so that I can have a look?
Thanks in advance!

**Comment by mushketyk** [comment]
Hi @kl0u,

Sure. I'll rebase it today.

**Comment by mushketyk** [comment]
Thank you for looking into this!

**Comment by mushketyk** [comment]
Hi @kl0u,

I've rebased the PR.

**Comment by kl0u** [comment]
Thanks a lot @mushketyk !

**Comment by kl0u** [comment]
Hi @mushketyk could you close this PR? I also commented on the related JIRA.

**Comment by mushketyk** [comment]
Closed.

**Pull request closed without merging**